floodviz/utils.py

```diff
@@ -0,0 +1,82 @@
+import requests
+
+
+def parse_rdb(endpoint, params):
+    """
+        Takes a web service endpoint, a dictionary of key/value pairs for the query parameters
+        and returns all parsed data from the webservice.
+
+        ARGS:
+            endpoint - String specifiying the web service url
+            params - dictionary holding url parameters
+
+                possible keys:
+                    'sites' (mandatory) - List of string site_ids
+                    'site_query' (mandatory) - service specific string for url construction that targets the sites.
+                    'extra' - service specific string prepended to the format if needed.
+                    'start_date_query' - service specific string for url construction that targets the start date of data
+                    'start_date' (madatory with start_date_query) - String specifying the start date for data. (YYYY-MM-DD)
+                    'end_date_query' - service specific string for url construction that targets the end date of data
+                    'end_date' (madatory with end_date_query) - String specifying the end date for data. (YYYY-MM-DD)
+                    'site_status' - service specific string for url construction. Cannot be used with 'agency' key
+                    'agency' - service specific string for url construction. Cannot be used with 'site_status' key
+
+        RETURNS:
+            url - url to request for data
+    """
+    all_data = []
+    sites = None
+    url = endpoint
+    data_format = '?format=rdb'
+    if params.get('sites') is not None:
```

Comment:
@mbucknell, I think I understand what you are getting at here. Similar to 'passing parameters to url' at this [link?](http://docs.python-requests.org/en/master/user/quickstart/)